main.py

```diff
@@ -39,6 +38,7 @@ def process_annual_weather_extremes(self, year):
                 weather_data.append(WeatherData.from_csv_row(row))
 
         # Filter out None values that might have been returned by from_csv_row
+
         weather_data = [data for data in weather_data if data is not None]
```

Comment:
Why not just do all this post processing of data like converting to WeatherData and checking nulls inside FileReader class? Because you would need it in other methods as well